Azure/azure-service-operator
# Fix to handle FailoverGroupDnsRecordInUse

Fixes #755 

This fix will log the error to the object status:
```
Status:
  Failed Provisioning:  true
  Message:              Reconcilation hit unrecoverable error sql.FailoverGroupsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidFailoverGroupRegion" Message="Secondary server specified in a Failover Group cannot reside in the same region."
  Requested:            2020-03-18T20:21:54Z
Events:
  Type    Reason      Age                From                              Message
  ----    ------      ----               ----                              -------
  Normal  Added       52s                AzureSqlFailoverGroup-controller  Object finalizer is added
  Normal  Reconciled  46s (x2 over 49s)  AzureSqlFailoverGroup-controller  Successfully reconciled
```

To test:
Create 2 sql servers and databases
Create sql failover group with default name in sample yaml
Check sql failover spec to ensure this error was logged and handled

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**How does this PR make you feel**:
![gif](https://giphy.com/)

**If applicable**:
